### DISCUSSION TITLE
GoogleAdsHook: enable OAuth Desktop and Web Application Flows

### DISCUSSION THREAD

**Discussion by LeonGraveland** [apache/airflow#30954] [Ideas]
Hi team, 

Currently the GoogleAdsHook is supporting authentication using the [OAuth Service Account Flow](https://developers.google.com/google-ads/api/docs/client-libs/python/oauth-service). 

Would it make sense to support the [OAuth Desktop and Web Application Flows](https://developers.google.com/google-ads/api/docs/client-libs/python/oauth-web)  as well? 

Asking since I've been working on connecting with google-ads, but having the credentials generated using the OAuth Desktop and Web Application Flow. Meaning my credentials look like:
`{"developer_token":"",
"refresh_token":"",
"client_id":"",
"client_secret":""} 
`

Happy to give it a go and create a PR, but just wanted to be sure this isn't a malpractice.

**Comment by pankajastro** [comment]
I think makes sense to have this if not supported. A PR would be great.

**Reply by LeonGraveland** [reply]
Sorry for the late reply, a year went by and I finally create a draft PR https://github.com/apache/airflow/pull/37417
As a first contributor, it is not clear how to link this discussion to the PR, should I first create an issue from this discussion?

**Reply by potiuk** [reply]
No need - you can refer to the discussion by including link to it.

**Reply by LeonGraveland** [reply]
Thanks for the swift response, then we're all good already
